types/src/transaction/transaction_target.rs

```diff
@@ -23,6 +21,144 @@ use rand::{Rng, RngCore};
+        ///
+        /// This is for protection against phishing attack where a malicious session code drains
+        /// the balance of the caller account. The amount stated here is the maximum amount
+        /// that can be transferred from the caller account to the session account.
+        transferred_value: u64,
+        /// The seed for the session code that is used for an installer.
+        seed: Option<[u8; 32]>,
+    },
+}
+
+impl TransactionRuntimeParams {
+    /// Returns the contract runtime tag.
+    pub fn contract_runtime_tag(&self) -> ContractRuntimeTag {
+        match self {
+            TransactionRuntimeParams::VmCasperV1 => ContractRuntimeTag::VmCasperV1,
+            TransactionRuntimeParams::VmCasperV2 { .. } => ContractRuntimeTag::VmCasperV2,
+        }
+    }
+
+    pub fn seed(&self) -> Option<[u8; 32]> {
+        match self {
+            TransactionRuntimeParams::VmCasperV1 => None,
+            TransactionRuntimeParams::VmCasperV2 { seed, .. } => *seed,
+        }
+    }
+}
+
+impl ToBytes for TransactionRuntimeParams {
+    fn to_bytes(&self) -> Result<Vec<u8>, Error> {
+        match self {
+            TransactionRuntimeParams::VmCasperV1 => {
+                CalltableSerializationEnvelopeBuilder::new(vec![
+                    crate::bytesrepr::U8_SERIALIZED_LENGTH,
+                ])?
+                .add_field(TAG_FIELD_INDEX, &VM_CASPER_V1_TAG)?
+                .binary_payload_bytes()
+            }
+            TransactionRuntimeParams::VmCasperV2 {
+                transferred_value,
+                seed,
+            } => CalltableSerializationEnvelopeBuilder::new(vec![
+                crate::bytesrepr::U8_SERIALIZED_LENGTH,
+                transferred_value.serialized_length(),
+                seed.serialized_length(),
+            ])?
+            .add_field(TAG_FIELD_INDEX, &VM_CASPER_V2_TAG)?
+            .add_field(TRANSFERRED_VALUE_INDEX, transferred_value)?
+            .add_field(SEED_VALUE_INDEX, seed)?
+            .binary_payload_bytes(),
+        }
+    }
+
+    fn serialized_length(&self) -> usize {
+        match self {
+            TransactionRuntimeParams::VmCasperV1 => {
+                CalltableSerializationEnvelope::estimate_size(vec![
+                    crate::bytesrepr::U8_SERIALIZED_LENGTH,
+                ])
+            }
+            TransactionRuntimeParams::VmCasperV2 {
+                transferred_value,
+                seed,
+            } => CalltableSerializationEnvelope::estimate_size(vec![
+                crate::bytesrepr::U8_SERIALIZED_LENGTH,
+                transferred_value.serialized_length(),
+                seed.serialized_length(),
+            ]),
+        }
+    }
+}
+
+impl FromBytes for TransactionRuntimeParams {
+    fn from_bytes(bytes: &[u8]) -> Result<(TransactionRuntimeParams, &[u8]), Error> {
+        let (binary_payload, remainder) = CalltableSerializationEnvelope::from_bytes(4, bytes)?;
```

Comment:
The VmCasperV2 variant has 2 fields so - including the enum tag - we should expect at most 3 fields here?

Reply:
Addressed